proto/viam/provisioning/provisioning.proto

```diff
@@ -0,0 +1,90 @@
+syntax = "proto3";
+
+package viam.provisioning.v1;
+
+option go_package = "go.viam.com/api/provisioning/v1";
+
+// ProvisionService is used by viam-agent to retrieve device and robot info.
+service ProvisionService {
+  // GetDeviceStatus is for retrieving the status of the device including networking.
+  rpc GetDeviceStatus(GetDeviceStatusRequest) returns (GetDeviceStatusResponse);
+
+  // SetWifiCredentials is to set the wifi credentials.
+  rpc SetWifiCredentials(SetWifiCredentialsRequest) returns (SetWifiCredentialsResponse);
```

Comment:
I'm going back and forth about having an AgentNetworking service and moving the wifi related functions there since networking management will be available outside of provisioning in the future. WDYT?

Reply:
I think they can use the same service. It just won't use the "SetRobotCredentials" call.